openfe/setup/network.py

```diff
@@ -1,13 +1,64 @@
 import math
+import networkx as nx
+from rdkit import Chem
+from typing import TypeVar
+
+
+NetworkXGraph = TypeVar("NetworkXGraph")
+
+
+def _hashmol(mol):
+    # canonical by default
+    # TODO: coordinates
+    return Chem.MolToSmiles(mol)
 
 
 class Network:
+    """A set of ligands and mappings between them"""
+    _graph: NetworkXGraph
+
+    def __init__(self):
+        self._graph = nx.Graph()
+
+    def __getitem__(self, key):
+        key = (_hashmol(key[0]), _hashmol(key[1]))
+
+        return self._graph[key]
+
+    def add_edge(self, atommapping):
+        """Manually add an edge onto an existing Network
+
+        Parameters
+        ----------
+        atommapping : AtomMapping
+          the mapping between two ligands to add
+
+        Raises
+        ------
+        ValueError
+          if an edge between two ligands already exists
+        """
+        key = (_hashmol(atommapping.mol1), _hashmol(atommapping.mol2))
+
+        if key in self._graph:
+            raise ValueError("This edge already exists in Network")
+
+        self._graph.add_edge(key, atommapping)
+
+    def get_edges(self):
+        """Iterable of edges in this Network"""
+        yield from self._graph.edges()
+
     @classmethod
-    def from_lomap(self, lomap_file):
-        pass
+    def from_lomap(cls, lomap_file):
+        raise NotImplementedError  # issue #16
+
+    def to_json(self):
+        """Return json string representation of this Network"""
+        raise NotImplementedError  # issue #15
```

Comment:
Suggestion: serialize by converting to json-serializable dict, not directly to json. This makes created mutated versions of the class very easy. We probably need some more discussion on serialization, generally.

Also, I'm not sure that JSON is the best serialization format here. Much as I hate XML, there are multiple standard file formats for serializing graphs (with arbitrary metadata) as XML.